x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -157,6 +225,123 @@ export function XyToolbar(props: VisualizationToolbarProps<State>) {
+              idToSelectedMap={tickLabelsVisibilitySettings}
+              onChange={(id) => onTickLabelsVisibilitySettingsChange(id)}
+              buttonSize="compressed"
+              isFullWidth
+              type="multi"
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={i18n.translate('xpack.lens.xyChart.Gridlines', {
+              defaultMessage: 'Gridlines',
+            })}
+          >
+            <EuiButtonGroup
+              name="lnsGridlines"
+              data-test-subj="lnsGridlinesSettings"
+              legend="Group of Gridlines Visibility Settings"
+              options={axes}
+              idToSelectedMap={gridlinesVisibilitySettings}
+              onChange={(id) => onGridlinesVisibilitySettingsChange(id)}
+              buttonSize="compressed"
+              isFullWidth
+              type="multi"
+            />
+          </EuiFormRow>
+          <EuiHorizontalRule margin="s" />
+          <EuiTitle size="xxs">
+            <span>
+              {i18n.translate('xpack.lens.xyChart.axisTitles', { defaultMessage: 'Axis titles' })}
+            </span>
+          </EuiTitle>
+          <EuiFormRow
+            display="columnCompressed"
+            label={
+              <EuiFlexGroup gutterSize="s">
+                <EuiFlexItem>X-axis</EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiSwitch
+                    data-test-subj="lnsshowXAxisTitleSwitch"
+                    showLabel={false}
+                    label={i18n.translate('xpack.lens.xyChart.showXAxisTitleLabel', {
+                      defaultMessage: 'show X-axis Title',
+                    })}
+                    onChange={({ target }) =>
+                      props.setState({ ...props.state, showXAxisTitle: target.checked })
+                    }
+                    checked={props.state?.showXAxisTitle ?? true}
+                  />
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            }
+          >
+            <EuiFieldText
+              data-test-subj="lnsXAxisTitle"
+              compressed
+              placeholder={i18n.translate('xpack.lens.xyChart.overwriteXaxis', {
+                defaultMessage: 'Overwrite X-axis title',
+              })}
+              value={xtitle}
+              disabled={
+                props.state && 'showXAxisTitle' in props.state ? !props.state.showXAxisTitle : false
+              }
+              onChange={({ target }) => onXTitleChange(target.value)}
+              aria-label={i18n.translate('xpack.lens.xyChart.overwriteXaxis', {
+                defaultMessage: 'Overwrite X-axis title',
+              })}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={
+              <EuiFlexGroup gutterSize="s">
+                <EuiFlexItem>Y-axis</EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiSwitch
```

Comment:
```suggestion
                  <EuiSwitch
                    compressed
```

Reply:
These patterns may also need to be evaluated by screenreaders to ensure they're read properly.